src/validation.ts

```diff
@@ -229,6 +241,73 @@ export class OpenAPIValidator {
     return result;
   }
 
+  /**
+   * Validates a response against a prebuilt Ajv validator and returns the result
+   *
+   * @param {*} headers
+   * @param {(Operation | string)} [operation]
+   * @param {number} [statusCode]
+   * @param {string} [setMatchType] - one of 'any', 'superset', 'subset', 'exact'
+   * @returns {ValidationResult}
+   * @memberof OpenAPIRequestValidator
+   */
+  public validateResponseHeaders(headers: any, operation: Operation | string, statusCode: number, setMatchType?: string)
+    : ValidationResult {
+    const result: ValidationResult = {
+      valid: true,
+      errors: [],
+    };
+
+    if (typeof operation === 'string') {
+      operation = this.router.getOperation(operation);
+    }
+
+    if (!operation || !operation.operationId) {
+      throw new Error(`Unknown operation`);
+    }
+
+    if (!setMatchType) {
+      setMatchType = this.allowedSetMatchTypes[0];
+    } else if (!_.includes(this.allowedSetMatchTypes, setMatchType)) {
+      throw new Error(`Unknown setMatchType ${setMatchType}`);
+    }
+
+    const { operationId } = operation;
+    const validateMap: ResponseHeadersValidateFunctionMap = this.getResponseHeadersValidatorForOperation(operationId);
+
+    if (validateMap) {
+      let strStatusCode = statusCode.toString();
+      let validate = validateMap[strStatusCode] && validateMap[strStatusCode][setMatchType];
+
+      if (!validate) {
+        // The spec allow statusCode to be 1XX, 2XX ...
+        strStatusCode = Math.floor(statusCode / 100) + 'XX';
+        validate = validateMap[strStatusCode] && validateMap[strStatusCode][setMatchType];
+      }
+
+      if (!validate) {
+        strStatusCode = 'default';
+        validate = validateMap[strStatusCode] && validateMap[strStatusCode][setMatchType];
+      }
+
+      if (validate) {
+        validate({headers});
```

Comment:
We need to map keys of `headers` to lowercase them first to make sure headers are handled non-case-sensitive.